src/throttler/throttler.ts

```diff
@@ -31,7 +31,13 @@ export interface ThrottlerStats {
   elapsed: number;
 }
 
-export abstract class Throttler<T> {
+interface TaskData<T, R> {
+  task: T;
+  retryCount: number;
+  wait?: { resolve: (R: any) => void; reject: (err: Error) => void };
```

Comment:
should this be:

```suggestion
  wait?: { resolve: (value: R) => void; reject: (err: Error) => void };
```

?

Reply:
Sorry, that was a typo.